publication-event-handlers/src/test/java/no/unit/nva/publication/events/handlers/batch/EventBasedBatchScanHandlerTest.java

```diff
@@ -0,0 +1,211 @@
+        throws TransactionFailedException {
+        createRandomResources(5);
+        handler.handleRequest(createInitialScanRequest(ONE_ENTRY_PER_EVENT), output, context);
+
+        var expectedStaringPointForNextEvent = getLatestEmittedStartingPoint();
+        var secondScanRequest = new ScanDatabaseRequest(ONE_ENTRY_PER_EVENT, expectedStaringPointForNextEvent);
+        handler.handleRequest(eventToInputStream(secondScanRequest), output, context);
+
+        var scanStartingPointSentToTheService =
+            scanningStartingPoints.get(scanningStartingPoints.size() - 1);
+
+        assertThat(scanStartingPointSentToTheService, is(equalTo(expectedStaringPointForNextEvent)));
+    }
+
+    @Test
+    void shouldNotGoIntoInfiniteLoop() throws TransactionFailedException {
+        createRandomResources(20);
+        pushInitialEntryInEventBridge(new ScanDatabaseRequest(ONE_ENTRY_PER_EVENT, START_FROM_BEGINNING));
+        while (thereAreMoreEventsInEventBridge()) {
+            var currentRequest = consumeLatestEmittedEvent();
+            handler.handleRequest(eventToInputStream(currentRequest), output, context);
+        }
+        assertThat(eventBridgeClient.getRequestEntries(), is(empty()));
+    }
+
+    @Test
+    void shouldLogFailureWhenExceptionIsThrown() {
+        final var logger = LogUtils.getTestingAppenderForRootLogger();
+        var expectedExceptionMessage = randomString();
+        var spiedResourceService = spy(resourceService);
+        doThrow(new RuntimeException(expectedExceptionMessage)).when(spiedResourceService)
+            .scanResources(anyInt(), any());
+
+        handler = new EventBasedBatchScanHandler(spiedResourceService, eventBridgeClient);
+        Executable action = () -> handler.handleRequest(createInitialScanRequest(ONE_ENTRY_PER_EVENT), output, context);
+        assertThrows(RuntimeException.class, action);
+        assertThat(logger.getMessages(), containsString(expectedExceptionMessage));
+    }
+
+    private void createRandomResources(int numberOfResources) throws TransactionFailedException {
+        for (int i = 0; i < numberOfResources; i++) {
+            resourceService.createPublication(PublicationGenerator.randomPublication());
+        }
+    }
+
+    private ResourceService mockResourceService(AmazonDynamoDB dynamoDbClient) {
+        return new ResourceService(dynamoDbClient, clock) {
+            @Override
+            public ListingResult<ResourceUpdate> scanResources(int pageSize, Map<String, AttributeValue> startMarker) {
+                if (nonNull(startMarker)) {
+                    scanningStartingPoints.add(startMarker);
+                }
+                return super.scanResources(pageSize, startMarker);
+            }
+        };
+    }
+
+    private FakeContext mockContent() {
+        return new FakeContext() {
+            @Override
+            public String getInvokedFunctionArn() {
+                return randomString();
+            }
```

Comment:
FakeContext throw now UnsupportedException and we should just fix it to return a randomString.